### PR TITLE
Rule idea: Focusable element has visible focus (bd5de3)

### PR DESCRIPTION
**Pull request by annethyme**
Looking for input on whether this rule is actually testing a WCAG requirement + if people agree on the general direction of the current draft.

Closes issue: none

## Guidance for the PR (pull request) creator

# How to Review And Approve
- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines. 
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.

### CODE REVIEW

**Comment by annethyme** [comment]
Only just realised that @EmmaJP and @john-urbilog has already started up a composite rule like this in https://github.com/auto-wcag/auto-wcag/issues/339.

I don't think it is necessary to do it as a composite rule, but let's get some input :)

File: _rules/bd5de3.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+- Anne Thyme Nørregaard
+---
+
+## Applicability
+
+This rule applies to all elements that can receive focus.
+
+## Expectation
+
+When the target element's focus state is triggered, a [visible](#visible) change in text style, border, box-shadow, background-image or color occurs for the element.
```
Comment (WilcoFiers) [inline _rules/bd5de3.md]: I don't think you can use the "visible" definition here. I'm also not sure these are all the CSS properties to look for, but I think this is a good direction. You may want to consider leaving just allowing any style change to indicate focus. Not sure what the best way to do this is.

Comment (ShadowBB) [inline _rules/bd5de3.md]: I think it is better to allow any style change to indicate focus, not just these specific examples. For instance a menu could show the focused item by making the text larger or indenting the menu item that has received focus, making it stand out from it's vertically aligned siblings.

File: _rules/bd5de3.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+- DOM Tree
+- CSS Styling
+
+ authors:
+- Anne Thyme Nørregaard
+---
+
+## Applicability
+
+This rule applies to all elements that can receive focus.
```
Comment (ShadowBB) [inline _rules/bd5de3.md]: This needs to specify somehow that elements with a tabindex="-1" which can still receive focus with the help of javascript are still applicable. I am unsure whether or not such an applicability can be tested automatically.

**Comment by kasperisager** [comment]
An interesting case to consider is "custom" radio and check boxes as popularized by Bootstrap: https://getbootstrap.com/docs/4.3/components/forms/#custom-forms. These consist of a visually hidden `<input>` element and an adjacent `<label>` element that is highlighted when the `<input>` element receives focus. As such, the element itself has no visible focus but an adjacent element has.

File: pages/glossary/change-in-appearance.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+title: Change in appearance
+key: change-in-appearance
+---
+
+A change in appearance for an element means that when comparing the element before and after the possible change has happened, the rendered pixels have changed within an area that includes the element plus a 50 pixel margin to each side.
```
Comment (WilcoFiers) [inline pages/glossary/change-in-appearance.md]: You should probably consider timing for this as well. Is this change in appearance is supposed to happen in the first frame paint after the change? If not, should there be a delay?

**Comment by ShadowBB** [comment]
Oh, and for clarity sake: I do think this tests a WCAG requirement and I agree with the general direction.

File: _rules/bd5de3.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
+
+---
+id: edb7b7
+name: Focusable element has visible focus
+test_type: atomic
+
+ description: |
+  This rule checks that all elements that can receive focus has a focus styling that is visibly different from the normal styling.
```
Comment (jkodu) [inline _rules/bd5de3.md]: editorial: visibly > visually

File: _rules/bd5de3.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
+
+---
+id: edb7b7
+name: Focusable element has visible focus
+test_type: atomic
```
Comment (jkodu) [inline _rules/bd5de3.md]: change to `rule_type`

File: _rules/bd5de3.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+ description: |
+  This rule checks that all elements that can receive focus has a focus styling that is visibly different from the normal styling.
+  
+ success_criterion: 
+- 2.4.7 # Focus Visible
+
+ test_aspects:
```
Comment (jkodu) [inline _rules/bd5de3.md]: change to `input aspects`

File: _rules/bd5de3.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+---
+id: edb7b7
+name: Focusable element has visible focus
+test_type: atomic
+
+ description: |
+  This rule checks that all elements that can receive focus has a focus styling that is visibly different from the normal styling.
+  
+ success_criterion: 
+- 2.4.7 # Focus Visible
```
Comment (jkodu) [inline _rules/bd5de3.md]: believe this is `accessibility_requirements` now

**Comment by Jym77** [comment]
Note to self:
Some elements might auto-steal focus of some others. Typically, a modal windows does not really prevent background elements to receive focus but steal it as soon as focus is given to something out of the modal. Thus, rule has to trigger on having focus, not on receiving it.

**Comment by jkodu** [comment]
based on ACT meeting on 17 October 2019, it was agreed that this PR will be closed until we have an active person to work on it.

**Pull request closed without merging**